crates/bevy_ecs/src/change_detection.rs

```diff
@@ -65,6 +72,25 @@ pub trait DetectChanges {
     /// [`SystemChangeTick`](crate::system::SystemChangeTick)
     /// [`SystemParam`](crate::system::SystemParam).
     fn last_changed(&self) -> u32;
+
+    /// Sets `self` to `value`, if and only if `*self != *value`
+    ///
+    /// `T` is the type stored within the smart pointer (e.g. [`Mut`] or [`ResMut`]).
+    ///
+    /// This is useful to avoid accidentally triggering change detection when no change is made,
+    /// as changes are usually deemed to be made when [`DerefMut`] is used.
```

Comment:
```suggestion
    /// This is useful to ensure change detection is only triggered when the underlying value 
    /// changes, instead of every time [`DerefMut`] is used.
```
Trying to add clarity by removing the double negative.